test/compressed.js

```diff
@@ -0,0 +1,49 @@
+var test = require('tape');
+var zlib = require('zlib');
+var Graylog = require('..');
+
+test('compressed', function (t) {
+	var client = new Graylog({ servers: [{ host: '127.0.0.1', port: 12345 }], deflate: 'optimal' });
+
+	client.on('message', function (message) {
+		message = JSON.parse(zlib.inflateSync(message));
+
+		t.equal(message.short_message, 'short message');
+		t.equal(message.full_message, 'full message'.repeat(1000));
```

Comment:
You could store `'full message'.repeat(1000)` in a variable instead of regenerating it.